onnxruntime/test/providers/cuda/cuda_mempool_arena_test.cc

```diff
@@ -65,9 +73,10 @@ static bool IsCudaMemPoolSupported() {
   cudaMemPool_t pool;
   auto cuda_error = cudaMemPoolCreate(&pool, &props);
   if (cuda_error != cudaSuccess) {
+    ClearCudaError();
```

Comment:
Should we peek at the error before trying to create the mempool to check if there were issues from running other tests **before** trying to attempt mem pool creation and subsequently resetting the error if mempool creation failed?

Reply:
The rationale here is as follows: 1) Typical CUDA code makes use of a macro that clears the error automatically. 2) the issue did not occur before my check-in where I use cuda RT directly, and the subsequent test started failing then.

I may still be wrong, but I think this is what is happening now.